### PR TITLE
Open (minting) fee per krAsset

### PR DESCRIPTION
**Pull request by diofinance**
This PR introduces a per krAsset type 'open fee' assessed when opening krAsset debt positions that operates similarly to the close fee. This PR includes a `calcExpectedFee` function that returns the amount of collateral expected to be paid for a specified krAsset, krAsset amount, and fee assessment type (mint or burn).

The `MAX_CLOSE_FEE` and `MAX_OPEN_FEE` are currently hardcoded to 10%, we should discuss if these limits are reasonable.

### CODE REVIEW

File: src/contracts/minter/facets/AccountStateFacet.sol [hunk 1]
```diff
@@ -143,4 +149,84 @@ contract AccountStateFacet is IAccountState {
...
+
+        // Calculate the value of the fee according to the value of the krAsset
+        FixedPoint.Unsigned memory feeValue = FixedPoint
+            .Unsigned(uint256(krAsset.oracle.latestAnswer()))
+            .mul(FixedPoint.Unsigned(_kreskoAssetAmount))
+            .mul(Fee(_feeType) == Fee.Open ? krAsset.openFee : krAsset.closeFee);
+
+        address[] memory accountCollateralAssets = ms().depositedCollateralAssets[_account];
+
+        ExpectedFeeRuntimeInfo memory info; // Using ExpectedFeeRuntimeInfo struct to avoid StackTooDeep error
```
Comment (panukettu) [inline src/contracts/minter/facets/AccountStateFacet.sol:181]: 👍

File: src/contracts/minter/libs/LibMint.sol [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import {Arrays} from "../../libs/Arrays.sol";
+import {MinterEvent} from "../../libs/Events.sol";
+import {FixedPoint} from "../../libs/FixedPoint.sol";
+import {Math} from "../../libs/Math.sol";
+
+import {LibCalc} from "./LibCalculation.sol";
+import {KrAsset} from "../MinterTypes.sol";
+import {MinterState} from "../MinterState.sol";
+
+library LibMint {
```
Comment (panukettu) [inline src/contracts/minter/libs/LibMint.sol:16]: I think we could call this `LibFees` and move all fee stuff here? So close/open fees at this stage

**Review by panukettu — APPROVED** [review]
LGTM, can address the structural change or not

File: src/contracts/minter/libs/LibMint.sol (hunk 2 above)
Comment (diofinance) [reply src/contracts/minter/libs/LibMint.sol:16]: I agree, makes more sense to colocate the functions. We could combine them as the only difference is line 43 `krAsset.openFee` vs `krAsset.closeFee` and then line 69 `OpenFeePaid` vs `CloseFeePaid` and use Fee.Open/Fee.Close enum as a parameter...but it's not as clean, let's discuss on call today

**Comment by diofinance** [comment]
Messed up conflict resolution when using git's in-browser editor, will resolve the failed test after today's eng meeting

**Comment by diofinance** [comment]
Closed in favor of latest

**Pull request closed without merging**